stdlib/public/core/String.swift

```diff
@@ -849,14 +849,29 @@ extension String {
       }
     }
 
-    var result = ""
-    result.reserveCapacity(utf8.count)
+    var buffer: [UInt8] = []
+    buffer.reserveCapacity(utf8.count)
 
     for scalar in unicodeScalars {
-      result += scalar.properties.lowercaseMapping
+      buffer.withUnsafeMutableBytes { buffer in
+        switch scalar.properties._getMapping(.lowercase) {
+        case (let scalar?, nil):
+          scalar.withUTF8CodeUnits {
+            buffer.copyMemory(from: UnsafeRawBufferPointer($0))
```

Comment:
This is not how you are supposed to copy data in to an Array. You should not change the initialized count inside `wihUnsafeMutableBytes` (also, is this overwriting the same location in a loop? Where is the offset being tracked?).

If you know the exact capacity, use `String(unsafeUninitializedCapacity:initializer:)` and save a copy. Otherwise, use `append(contentsOf:)`